### PR TITLE
fix(treesitter): clip end row early 

### PR DESCRIPTION
**Pull request by tomtomjhj**
Also add some more argument checks.

Ref: https://github.com/neovim/neovim/pull/28417#issuecomment-2068319504
cc @justinmk

### CODE REVIEW

File: src/nvim/lua/stdlib.c [hunk 1]
```diff
@@ -549,19 +549,21 @@ static int nlua_iconv(lua_State *lstate)
...
-  Error err = ERROR_INIT;
-  win_T *win = find_window_by_handle(window, &err);
-  if (ERROR_SET(&err)) {
-    nlua_push_errstr(lstate, err.msg);
-    api_clear_error(&err);
-    lua_error(lstate);
-    return 0;
+  win_T *win = handle_get_window(window);
+  if (!win) {
+    return luaL_error(lstate, "invalid window");
```
Comment (justinmk) [inline src/nvim/lua/stdlib.c:554]: if it's a false positive (i.e. "can never happen"), assert() or abort() is fine too

Comment (tomtomjhj) [reply src/nvim/lua/stdlib.c:554]: `find_window_by_handle` in the old version uses `handle_get_window` and does null check internally, and sets `err` in that case. So `ERROR_SET(&err)` in the old version implies `win` is non-null. But adding assert to that is somewhat redundant, so I replaced those stuff to `handle_get_window` and direct null check.

**Comment by tomtomjhj** [comment]
With the additional argument checks, I noticed a regression from #28417. cc @lewis6991

File: src/nvim/lua/stdlib.c (hunk 1 above)
Comment (ii14) [reply src/nvim/lua/stdlib.c:554]: It's redundant either way, because that warning was a false positive.

Comment (tomtomjhj) [reply src/nvim/lua/stdlib.c:554]: Right, but I just wanted to reduce the noise from the warning

**Pull request merged**